core/src/main/scala/com/nvidia/spark/rapids/tool/planparser/SQLPlanParser.scala

```diff
@@ -73,6 +74,56 @@ object SQLPlanParser extends Logging {
 
   val windowFunctionPattern = """(\w+)\(""".r
 
+  /**
+   * This function is used to create a set of nodes that should be skipped while parsing the Execs
+   * of a specific node.
+   * When a reused expression appears in a SparkPlan, the sparkPlanGraph constructed from the
+   * eventlog will have duplicates for all the ancestors of the exec (i.e., "ReusedExchange").
+   * This leads to a gap in the GPU speedups across different platforms which generate the graph
+   * without duplicates.
+   * A work around is to detect all the duplicates nodes so that we can mark them as "shouldRemove".
+   * If a wholeGen node has all the children labeled as ancestor of reused-exchange, then the
+   * wholeGen should also be added to the same set.
+   * @param planGraph the graph generated for a spark plan. This graph can be different depending on
+   *                  the spark-sql jar version used to construct the graph from existing eventlogs.
+   * @return a set of node IDS to be skipped during the aggregation of the speedups.
+   */
+  private def buildSkippedReusedNodesForPlan(planGraph: SparkPlanGraph): Set[Long] = {
+    def findNodeAncestors(planGraph: SparkPlanGraph,
+        graphNode: SparkPlanGraphNode): mutable.Set[Long] = {
+      // Given a node in the graph, this function is to go backward to find all the ancestors of
+      // the node including the node, itself.
+      val visited = mutable.Set[Long](graphNode.id)
+      val q1 = mutable.Queue[Long](graphNode.id)
+      while (q1.nonEmpty) {
+        val curNode = q1.dequeue()
+        val allSinkEdges = planGraph.edges
+          .filter(e => e.toId == curNode)
+          .filterNot(e => visited.contains(e.fromId))
+        for (currEdge <- allSinkEdges) {
+          q1.enqueue(currEdge.fromId)
+          visited += currEdge.fromId
+        }
+      }
+      // Loop on the wholeGen to see if any of them is covered by the ancestors path.
+      // This implies that the wholeStageCodeGen is also reused.
+      // Note that the following logic can be moved to the WholeStageCodegen parser. Handling the
+      // logic here has advantages:
+      //   1- no need to append to append to the final set
```

Comment:
Nit: 1- no need  to append to the final set

Reply:
Done!